pkgs/applications/video/freetube/default.nix

```diff
@@ -1,51 +1,109 @@
-{ stdenv, lib, fetchurl, appimageTools, makeWrapper, electron_22 }:
+{ stdenv
+, lib
+, fetchurl
+, undmg
+, appimageTools
+, makeWrapper
+, electron_22
+}:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "freetube does not support system: ${system}";
 
-stdenv.mkDerivation rec {
   pname = "freetube";
-  version = "0.19.0";
 
-  src = fetchurl {
-    url = "https://github.com/FreeTubeApp/FreeTube/releases/download/v${version}-beta/freetube_${version}_amd64.AppImage";
-    sha256 = "0yr5k9s3r4yvcx85bzwn6y4m03964ljnmhz7nf068zj87m9q8rcc";
-  };
+  x86_64-darwin-version = "0.19.0";
+  x86_64-darwin-sha256 = "85f9ea8db43b89082957f77821b3c7cdc48bb371b6f55f95158efc4c64fc9edc";
+
+  x86_64-linux-version = "0.19.0";
+  x86_64-linux-sha256 = "8c6584533d487e6480b3e7c36a2525260d50893796ff555067db933c749a257b";
+
+  aarch64-darwin-version = "0.19.0";
+  aarch64-darwin-sha256 = "85f9ea8db43b89082957f77821b3c7cdc48bb371b6f55f95158efc4c64fc9edc";
+
+  version = {
+    x86_64-darwin = x86_64-darwin-version;
+    x86_64-linux = x86_64-linux-version;
+    aarch64-darwin =  aarch64-darwin-version;
+  }.${system} or throwSystem;
+
+  src = let
+    base = "https://github.com/FreeTubeApp/FreeTube/releases/download";
+  in {
+    x86_64-darwin = fetchurl {
+      url = "${base}/v${version}-beta/freetube-${version}-mac-x64.dmg";
+      sha256 = x86_64-darwin-sha256;
+    };
+    x86_64-linux = fetchurl {
+      url = "${base}/v${version}-beta/freetube_${version}_amd64.AppImage";
+      sha256 = x86_64-linux-sha256;
+    };
+    aarch64-darwin = fetchurl {
+      url = "${base}/v${version}-beta/freetube-${version}-mac-x64.dmg";
+      sha256 = aarch64-darwin-sha256;
+    };
+  }.${system} or throwSystem;
 
-  appimageContents = appimageTools.extractType2 {
-    name = "${pname}-${version}";
-    inherit src;
+  meta = with lib; {
+    description = "An Open Source YouTube app for privacy";
+    homepage = "https://freetubeapp.io/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ ryneeverett alyaeanyx ];
+    platforms = [ "x86_64-darwin" "x86_64-linux" "aarch64-darwin" ];
   };
 
-  dontUnpack = true;
-  dontConfigure = true;
-  dontBuild = true;
+  linux = stdenv.mkDerivation rec {
```

Comment:
It's common practice to use `if` cases to distinguish between different platforms instead of defining attrsets, e.g. `installPhase = if stdenv.isDarwin then x else y`